### PR TITLE
Update dsdl submodule to specify legacy-v0 branch

### PR DESCRIPTION
**Pull request by aroberts109**
Pull request to address PX4/Firmware Issue #14336.

### CODE REVIEW

File: libuavcan_drivers/stm32/driver/src/internal.hpp [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #elif UAVCAN_STM32_NUTTX
 # include <nuttx/arch.h>
 # include <arch/board/board.h>
-# include <chip/stm32_tim.h>
+#include <stm32_tim.h>
```
Comment (pavel-kirienko) [inline libuavcan_drivers/stm32/driver/src/internal.hpp:14]: Why is this change needed here?

Comment (TSC21) [reply libuavcan_drivers/stm32/driver/src/internal.hpp:14]: I think this is a remain of something else. @ARob109 I think you applied your changes over a commit which is not the same HEAD as master. Please do a git rebase first.

Comment (aroberts109) [reply libuavcan_drivers/stm32/driver/src/internal.hpp:14]: Sorry about that. Still new to the process, got it rebased and issued new pull request.

Comment (pavel-kirienko) [reply libuavcan_drivers/stm32/driver/src/internal.hpp:14]: Thank you. For your future reference, you don't need to open a new PR after every change. You could have just pushed to this branch and GitHub would have updated the current PR automatically.
